addons/portal/controllers/portal.py

```diff
@@ -158,6 +159,15 @@ def _prepare_portal_layout_values(self):
             'page_name': 'home',
         }
 
+    def _check_page_visibility(self, key):
+        if hasattr(request, 'website'):
+            domain = AND([[("key", "=", key)], request.website.website_domain(request.website.id)])
+            record = request.env['ir.ui.view'].with_context(active_test=False).sudo().search(domain)
+            record_with_website = record.filtered(lambda x: x.website_id)
+            record_without_website = record - record_with_website
+            return record_with_website.active if record_with_website else record_without_website.active
+        return True
+
```

Comment:
@adsh-odoo, 

It could be placed before `_prepare_portal_layout_values` method.